.github/workflows/lint-and-test.yml

```diff
@@ -138,6 +138,45 @@ jobs:
         if: steps.check-scripts.outputs.skip != 'true' && steps.check-scripts.outputs.skip-test != 'true'
         run: cd ${{ matrix.package }} && yarn test
 
+  lint-knip:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout the PR branch
+        uses: actions/checkout@v2
+
+      - name: Get the base branch
+        id: base_branch
+        run: echo "::set-output name=branch::$(echo '${{ github.base_ref }}' || echo 'main')"
```

Comment:
You can skip this step if you do 

```
ref: ${{ github.base_ref }}
```

below. Not sure if there needs to be a default. Maybe not? If yes, then `develop` is probably best bet

Reply:
https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/#examples